tests/console/yast2_samba.pm

```diff
@@ -38,13 +38,11 @@ sub run {
         record_soft_failure 'bsc#1064405';
         send_key 'alt-c';
     }
-    send_key 'alt-f';
-
-    assert_screen([qw(yast2_ldap_configuration_general-setting_firewall yast2_still_susefirewall2)], 60);
-    if (match_has_tag 'yast2_still_susefirewall2') {
-        record_soft_failure 'bsc#1064405';
-        send_key 'alt-e';
+    # only older version like sles 12, tw, leap should still check the needle
+    if ((is_sle && !sle_version_at_least('15')) || is_leap && !leap_version_at_least('15.0')) {
```

Comment:
Please, either remove wrapping parenthesis for both && conditions, or wrap is_leap && .... 
Code will work, as logical AND has precedence on OR, but it's inconsistent from style perspective.

Reply:
In the comment you mention "tw" but Tumbleweed should not activate that branch. I think that comment is wrong